tests/js/basic-tests.js

```diff
@@ -0,0 +1,284 @@
+                            listener: "gpii.test.couchdb.checkResponse",
+                            event:    "{supplementalReadRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{supplementalReadRequest}.nativeResponse", "{arguments}.0", 200, "{testCaseHolder}.options.expected.supplementalRead"]
+                        }
+                    ]
+                },
+                {
+                    name: "Testing deleting a single record from the 'sample' database.",
+                    type: "test",
+                    sequence: [
+                        // The record should exist before we delete it.
+                        {
+                            func: "{preDeleteRequest}.send",
+                            args: [{}]
+                        },
+                        // confirm that the record exists now and delete the latest revision.
+                        {
+                            listener: "gpii.tests.couchdb.basic.checkRecordAndStartDelete",
+                            event:    "{preDeleteRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{preDeleteRequest}.nativeResponse", "{arguments}.0", 200, "{that}.options.expected.beforeDelete", "{deleteRequest}"]
+                        },
+                        // The delete request should be successful.
+                        {
+                            listener: "gpii.test.couchdb.checkResponse",
+                            event:    "{deleteRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{deleteRequest}.nativeResponse", "{arguments}.0", 200, "{testCaseHolder}.options.expected.afterDelete"]
+                        },
+                        // The record should no longer exist after we delete it.
+                        {
+                            func: "{verifyDeleteRequest}.send",
+                            args: [{}]
+                        },
+                        {
+                            listener: "gpii.test.couchdb.checkResponse",
+                            event:    "{verifyDeleteRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{verifyDeleteRequest}.nativeResponse", "{arguments}.0", 404]
+                        }
+                    ]
+                },
+                {
+                    name: "Testing inserting a record into the 'sample' database.",
+                    type: "test",
+                    sequence: [
+                        // The record should not exist before we create it.
+                        {
+                            func: "{preInsertRequest}.send",
```

Comment:
Same as the last comment: is it possible to handle preInsert internally?

Reply:
This check is making sure that the data doesn't already exist, i.e. that nothing is left over from a previous run.  Actual inserts by real users will not require an initial lookup.